api-utils/apiUtils.ts

```diff
@@ -0,0 +1,75 @@
+    const teamId: string = process.env.BASE_TEAM_ID as string;
+    const getSpacesEndpoint: string = `https://api.clickup.com/api/v2/team/${teamId}/space`;
+
+    const getSpacesResponse: APIResponse = await request.get(getSpacesEndpoint, { headers: { Authorization: apiKey } });
+    const spaceId: string = (await getSpacesResponse.json()).spaces.filter((space) => space.name === spaceName)[0].id;
+
+    return spaceId;
+  }
+
+  public static async getFolderIdByName(request: APIRequestContext, spaceName: string, folderName: string): Promise<string> {
+    const apiKey: string = process.env.API_KEY as string;
+    const spaceId: string = await this.getSpaceIdByName(request, spaceName);
+    const getFoldersEndpoint: string = `https://api.clickup.com/api/v2/space/${spaceId}/folder`;
+
+    const getFoldersResponse: APIResponse = await request.get(getFoldersEndpoint, { headers: { Authorization: apiKey } });
+    const folderId: string = (await getFoldersResponse.json()).folders.filter((folder) => folder.name === folderName)[0].id;
+
+    return folderId;
+  }
+
+  public static async getListIdByName(request: APIRequestContext, spaceName: string, folderName: string, listName: string): Promise<string> {
+    const apiKey: string = process.env.API_KEY as string;
+    const folderId: string = await this.getFolderIdByName(request, spaceName, folderName);
+    const getListsEndpoint: string = `https://api.clickup.com/api/v2/folder/${folderId}/list`;
+
+    const getListsResponse: APIResponse = await request.get(getListsEndpoint, { headers: { Authorization: apiKey } });
+    const listId: string = (await getListsResponse.json()).lists.filter((list) => list.name === listName)[0].id;
+
+    return listId;
+  }
+
+  // Obtains id of a base list for task tests - Team Space/Projects/Project 1
+  public static async getTaskIdFromBaseList(request: APIRequestContext, taskName: string): Promise<string> {
+    // TODO: move base names to separate class or dictionary
+    const apiKey: string = process.env.API_KEY as string;
+    const baseSpaceName: string = "Team Space";
+    const baseFolderName: string = "Projects";
+    const baseListName: string = "Project 1";
+    const baseListId: string = await this.getListIdByName(request, baseSpaceName, baseFolderName, baseListName);
+    const getTasksEndpoint: string = `https://api.clickup.com/api/v2/list/${baseListId}/task`;
+
+    const getTasksResponse: APIResponse = await request.get(getTasksEndpoint, { headers: { Authorization: apiKey } });
+    try {
+      const taskId: string = (await getTasksResponse.json()).tasks.filter((task) => task.name === taskName)[0].id;
+      return taskId;
+    } catch (error) {
+      if (error instanceof TypeError) {
+        console.error("Task with given name was not found!", error.message);
+      } else {
+        console.error("Unexpected error occurred while trying to obtain an id!", error.message);
+      }
+      return "";
+    }
```

Comment:
Usunąłem zmianę `taskName` w trakcie testu. @marian51 Zerknij proszę na obsługę tego wyjątku - zostawiamy tak i później będziemy przebudowywać tę klasę czy od razu dodać takie bloki do wszystkich metod?

Wydaje mi się, że wystarczy już tutaj powtarzania się i lepiej zrobić refactor.

Reply:
Ten block `try-catch`, jeżeli wystąpi błąd, to wypisze go w konsoli, ale nie zatrzyma działania programu? W sensie, jak nie znajdzie `id`, to zwraca pusty string? 

Obsługa errora spoko, na ten moment nie mam pomysłu, jak zrobić to z innymi metodami - czy też tam porobić takie bloki, czy jakoś inaczej (ogólniej) to ująć.

Reply:
Tak, z tym blokiem testy przechodzą.

Bez tego `try-catch` testy, w których jeden z tasków nie zostanie znaleziony nie przejdą na zielono. Na ten moment zostawię to tak i założę issue na posprzątanie tej klasy (na razie jeszcze nie wiem jak to zrobić tak, żeby było elegancko)